test-packages/e2e-tests/test/request-builder.spec.ts

```diff
@@ -26,7 +26,7 @@ async function createEntity(key: number) {
     .stringProperty('someValue')
     .dateProperty(moment(0))
     .timeOfDayProperty({ hours: 1, minutes: 2, seconds: 3 })
-    .dataTimeOffsetDataTimeProperty(moment(0))
+    // .dataTimeOffsetDataTimeProperty(moment(0))
```

Comment:
This throws the error:

> "message":"Value 2024-11-15T14:10:32.84Z is not a valid DateTime","target":"DataTimeOffsetDataTimeProperty","code":"400","@Common.numericSeverity":4

For DateTimeOffset, this is a valid format as per [OData docs]("message":"Value 2024-11-15T14:10:32.84Z is not a valid DateTime","target":"DataTimeOffsetDataTim
eProperty","code":"400","@Common.numericSeverity":4)

Reply:
→ see https://github.tools.sap/cap/issues/issues/17329#issuecomment-9011356

Reply:
@danjoa I tried to override the default mapping for `DateTime` property in the TestEntity model to have the correct precision:
```
  @odata: { Type: 'Edm.DateTimeOffset', Precision: 3 }
  DataTimeOffsetDataTimeProperty: DateTime;
```

The mock data in the db is also stored in the format: `2020-06-11T07:42:32.464Z`. The cap service still responds with 0 fractional digits. 

Any hint as to what we're missing here?

Reply:
The warning section [here](https://cap.cloud.sap/docs/advanced/odata#override-type-mapping) clarified the misunderstanding.